keps/sig-storage/1472-storage-capacity-tracking/README.md

```diff
@@ -585,34 +577,82 @@ instead. Obsolete objects need to be removed.
 
 To ensure that `CSIStorageCapacity` objects get removed when the driver
 deployment gets removed before it has a chance to clean up, each
-`CSIStorageCapacity` object needs an [owner
+`CSIStorageCapacity` object should have an [owner
 reference](https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#OwnerReference).
 
-For central provisioning, that has to be the Deployment or StatefulSet
-that defines the provisioner pods. That way, provisioning can
-continue seamlessly when there are multiple instances with leadership
-election.
-
-For deployment with a daemon set, making the individual pod the owner
-is better than the daemon set as owner because then other instances do
-not need to figure out when to remove `CSIStorageCapacity` objects
-created on another node. It is also better than the node as owner
-because a driver might no longer be needed on a node although the node
-continues to exist (for example, when deploying a driver to nodes is
-controlled via node labels).
-
-While external-provisioner runs, it needs to update and potentially
+The owner should be the higher-level app object which defines the provisioner pods:
+- the Deployment or StatefulSet for central provisioning
+- the DaemonSet for distributed provisioning
+
+This way, provisioning and pod scheduling can continue seamlessly when
+there are multiple instances with leadership election or the driver
+gets upgraded.
+
+This ownership reference is optional. It gets enabled with
+external-provisioner command line arguments when deploying the
+driver. Without it, administrators must manually delete
+`CSIStorageCapacity` objects after uninstalling a CSI driver. To
+simplify that and for efficient access to objects, external-provisioner
+sets some labels:
+- `app.kubernetes.io/instance`: the CSI driver name
+- `app.kubernetes.io/managed-by`: `external-provisioner` for central provisioning,
+  `external-provisioner-<node name>` for distributed provisioning
+
+With that, `kubectl delete csistoragecapacities -l
+app.kubernetes.io/instance=my-csi.example.com` will delete just the
+objects of that driver.
+
+It is possible to create `CSIStorageCapacity` manually. This may be
+useful to specify capacity of simulated nodes before the autoscaler
+actually creates them. external-provisioner must never delete those.
+In external-provisioner 2.0.0, this was achieved by checking the owner
+reference. Now that the owner reference is optional, it will get
+achieved by ignoring all objects that don't have the
+`app.kubernetes.io/managed-by` labels described above.
```

Comment:
What happens if such object was manually created before?
Will the provisioner create its own (effectively a second one) or will skip creating it? If the later, will it update the existing one?

Reply:
If we end up filtering the objects by labels, then we may miss objects that were manually created without the labels and the scheduler may have problems. Agree we should decide on how to handle this conflict.